modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/js/input_localized/InputLocalized.tsx

```diff
@@ -0,0 +1,346 @@
+	 * {@link journal-web/.../TranslationManager.tsx}'s
+	 * `handleSelectedLanguageIdChange` function which doesn't currently connect
+	 * with the `selectedLanguageIdAtom`.
+	 */
+	useEffect(() => {
+		const handleLocaleChanged = (event: any) => {
+			const selectedLanguageId = event.item.getAttribute('data-value');
+
+			setSelectedLanguageId(selectedLanguageId);
+		};
+
+		Liferay.on('inputLocalized:localeChanged', handleLocaleChanged);
+
+		return () => {
+			Liferay.detach(
+				'inputLocalized:localeChanged',
+				handleLocaleChanged as () => void
+			);
+		};
+	}, [setSelectedLanguageId]);
+
+	/**
+	 * When the `selectedLanguageId` is changed, this will autofill the value
+	 * from the `defaultLanguageId` into the editor if there is no translation
+	 * for it yet.
+	 */
+	useEffect(() => {
+		if (!autofillFromDefault) {
+			return;
+		}
+
+		setTranslations((currentTranslations) => {
+			const selectedValue =
+				currentTranslations?.[selectedLanguageId] || '';
+
+			if (selectedValue === '') {
+				const defaultValue =
+					currentTranslations?.[defaultLanguageId] || '';
+
+				if (defaultValue) {
+					return {
+						...(currentTranslations || {}),
+						[selectedLanguageId]: defaultValue,
+					} as Translations['translations'];
+				}
+			}
+
+			return currentTranslations;
+		});
+	}, [autofillFromDefault, defaultLanguageId, selectedLanguageId]);
+
+	const _handleBlur = useCallback(
+		(event: EventInfo, editor: TEditor) => {
+			if (onBlur) {
+				onBlur(event, editor);
+			}
+
+			if (
+				onBlurMethod &&
+				typeof (window as any)[onBlurMethod] === 'function'
+			) {
+				(window as any)[onBlurMethod](event, editor);
+			}
+		},
+		[onBlur, onBlurMethod]
+	);
+
+	const _handleChange = useCallback(
+		(event: EventInfo, editor: TEditor) => {
+			setTranslations(
+				(translations) =>
+					({
+						...translations,
+						[selectedLanguageId]: editor.getData(),
+					}) as Translations['translations']
+			);
+
+			if (onChange) {
+				onChange(event, editor);
+			}
+
+			if (
+				onChangeMethod &&
+				typeof (window as any)[onChangeMethod] === 'function'
+			) {
+				(window as any)[onChangeMethod](event, editor);
+			}
+		},
+		[onChange, onChangeMethod, selectedLanguageId]
+	);
+
+	const _handleFocus = useCallback(
+		(event: EventInfo, editor: TEditor) => {
+			if (onFocus) {
+				onFocus(event, editor);
+			}
+
+			if (
+				onFocusMethod &&
+				typeof (window as any)[onFocusMethod] === 'function'
+			) {
+				(window as any)[onFocusMethod](event, editor);
+			}
+		},
+		[onFocus, onFocusMethod]
+	);
+
+	return (
+		<div className="c-gap-2 d-flex">
+			<div className="w-100">
+				<ClassicEditor
+					config={{
+						...editorConfig,
+						language: {
+							content: selectedLanguageId?.split('_')[0],
+						},
+					}}
+					data={translations?.[selectedLanguageId] || ''}
+					key={selectedLanguageId}
+					onBlur={_handleBlur}
+					onChange={_handleChange}
+					onFocus={_handleFocus}
```

Comment:
[naming nit] No need for `_` prefix. At one point we were considering `on*` vs `handle*` for internal handlers, and the conclusion was they are both equally good, but in DXP `on` is used far more often. So for consistency, I'd call these `onBlur`, `onChange` and `onFocus`.

Reply:
Fair enough, this is really a preference up to a developer. I don't really mind either, just prefer `on` because of consistency. Just for the sake of the argument, there is never a conflict with prop name. In fact, the same name results in ability to shorthand code for prop setting.